### PR TITLE
Fix: improve IPC module file structure

### PR DESCRIPTION
**Pull request by filippobrizzi**
# Description
No logic change, just a file restructure with breaking changes:
* Move all the files related to zenoh specifics from `ipc` to `ipc/zenoh` 
* Updated the namespace of the classes moved to zenoh folder to `ipc::zenoh` 
* Renamed publisher subscriber filename to raw_publisher raw_subscriber to match the class name
* Add a new Subscriber template class to match the Publisher
  * This is mainly cosmetics but it is nice to have the type in the subscriber for readability
* Removed from the `examples`  module the ipc command line arguments and use the one defined in the ipc module

### CODE REVIEW

File: modules/ipc/src/zenoh/conversions.cpp [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+#include <vector>
+
+#include <fmt/core.h>
+#include <zenoh/api/bytes.hxx>
+#include <zenoh/api/id.hxx>
+#include <zenoh/api/timestamp.hxx>
+
+namespace heph::ipc::zenoh {
+namespace {
+[[nodiscard]] auto toChrono(uint64_t timestamp) -> std::chrono::nanoseconds {
```
Comment (lhruby) [inline modules/ipc/src/zenoh/conversions.cpp:23]: Add units to timestamp, or are these `nanoseconds`?

**Review by lhruby — APPROVED** [review]
Given you wrote there is no logic change, I assume no code was added, removed or altered, thus approved. If there was a changed, please point to the relevant sections.

**Review by floriantschopp — APPROVED** [review]

File: modules/ipc/src/zenoh/conversions.cpp (hunk 1 above)
Comment (filippobrizzi) [reply modules/ipc/src/zenoh/conversions.cpp:23]: it is not nanoseconds, it is a weird format described in the documentation linked below

Comment (lhruby) [reply modules/ipc/src/zenoh/conversions.cpp:23]: merci

**Pull request merged**